src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/GenericIdentitySecurityToken.cs

```diff
@@ -8,21 +8,38 @@
 
 namespace CoreWCF.IdentityModel.Tokens
 {
-    internal class GenericSecurityToken : SecurityToken
+    public class GenericIdentitySecurityToken : SecurityToken
     {
         private readonly string _id;
         private readonly DateTime _effectiveTime;
         private readonly DateTime _expirationTime;
         private readonly GenericIdentity _genericIdentity;
 
-        internal GenericSecurityToken(string name, string id)
+        public GenericIdentitySecurityToken(string name)
+            : this(name, SecurityUniqueId.Create().Value)
+        {
+        }
+        public GenericIdentitySecurityToken(GenericIdentity genericIdentity)
+            : this(genericIdentity, SecurityUniqueId.Create().Value)
+        {
+        }
+
+        public GenericIdentitySecurityToken(string name, string id)
         {
```

Comment:
I don't think we should have constructors which take a name. If someone only has a name, they can construct a GenericIdentity themselves with that name and pass that.